src/main/scala/mesosphere/marathon/upgrade/DeploymentActor.scala

```diff
@@ -88,7 +88,8 @@ private class DeploymentActor(
       eventBus.publish(status)
 
       val futures = step.actions.map { action =>
-        healthCheckManager.addAllFor(action.app) // ensure health check actors are in place before tasks are launched
+        // ensure health check actors are in place before tasks are launched
+        healthCheckManager.addAllFor(action.app, Seq.empty)
```

Comment:
the "ensure ... are in place" semantics depend on a side-affect of addAllFor. would prefer to see a MarathonHealthCheckManager `prepareFor(...)` or something (and maybe that impl would invoke addAllFor, hiding the impl detail from DeploymentActor). Would make for a cleaner interface between deployments and health checks.